src/compiler/factory/node_factory.rs

```diff
@@ -69,6 +69,41 @@ impl NodeFactory {
         base_factory.create_base_token_node(kind)
     }
 
+    pub fn create_token<TBaseNodeFactory: BaseNodeFactory>(
+        &self,
+        base_factory: &TBaseNodeFactory,
+        token: SyntaxKind,
+    ) -> Token {
+        let node = self.create_base_token(base_factory, token);
+        Token { _node: node }
+    }
+
+    fn create_base_expression<TBaseNodeFactory: BaseNodeFactory>(
+        &self,
+        base_factory: &TBaseNodeFactory,
+        kind: SyntaxKind,
+    ) -> BaseNode {
+        let node = self.create_base_node(base_factory, kind);
+        node
+    }
+
+    pub fn create_binary_expression<TBaseNodeFactory: BaseNodeFactory>(
+        &self,
+        base_factory: &TBaseNodeFactory,
+        left: Expression,
+        operator: Node,
+        right: Expression,
+    ) -> BinaryExpression {
+        let node = self.create_base_expression(base_factory, SyntaxKind::BinaryExpression);
+        let node = BinaryExpression {
+            _node: node,
+            left: Box::new(left),
+            operator_token: Box::new(operator),
+            right: Box::new(right),
+        };
```

Comment:
For something like this where the boxing is an implementation detail of `BinaryExpression`, I'd probably hide that behind a constructor so you can do `BinaryExpression::new(node, left, operator, right)`.